ruby/lib/rubocop/cop/inclusive_code.rb

```diff
@@ -52,11 +52,10 @@ def investigate(processed_source)
 
             @all_non_inclusive_words.each do |non_inclusive_word|
               allowed = @allowed[non_inclusive_word]
-              scan_regex = if allowed.blank?
-                             /(?=#{non_inclusive_word})/i
-                           else
-                             /(?=#{non_inclusive_word})(?!(#{@allowed[non_inclusive_word]}))/i
-                           end
+              scan_regex = /(?=#{non_inclusive_word})/i
+              if allowed.present?
+                line = line.gsub(/(#{allowed})/i){ |match| '*' * match.size }
```

Comment:
This could be called for `N lines in project` * `N configs with "allow" settings`. To avoid allocating so many string objects for the "*", it may be a good idea to extract a constant to take advantage of the frozen string literal:

```ruby
  ALLOWED_TERM_MASK_CHAR = '*'
```